corehq/apps/userreports/pillow.py

```diff
@@ -208,6 +211,100 @@ def _save_doc_to_table(self, domain, table, doc, eval_context):
             # remove it until the next bootstrap call
             self.table_adapters_by_domain[domain].remove(table)
 
+    def process_changes_chunk(self, pillow_instance, changes):
```

Comment:
Can this method be broken up into helpers? It's getting a bit hard to follow at ~100 lines including the internal function. I think it could be written as:


```python
def process_changes:
    relevant_changes = sorted([change for change in changes if change.domain in self.domains_with_adapters])
    changes_by_domain = # {domain: [change1,change2, ...], ...}
    for domain in changes_by_domain:
        self.process_changes_by_domain(domain, changes_by_domain[domain])

def process_changes_by_domain(domain, changes):
    for changes in changes_by_doc_store:
        self.process_changes_by_doc_store(doc_store, changes)
```

(I think that you're already processing them by domain/doc store, but I think this will make it a little clearer.)